portal-kernel/src/com/liferay/portal/kernel/dao/orm/IndexableActionableDynamicQuery.java

```diff
@@ -151,14 +151,14 @@ protected void sendStatusMessage(long documentIntervalCount) {
 
 	private static final long _STATUS_INTERVAL = 100;
 
-	private static volatile IndexWriterHelper _indexWriterHelperProxy =
-		ServiceProxyFactory.newServiceTrackedInstance(
-			IndexWriterHelper.class, IndexableActionableDynamicQuery.class,
-			"_indexWriterHelperProxy", false);
+	private static final Snapshot<IndexWriterHelper>
+		_indexWriterHelperProxySnapshot = new Snapshot<>(
+			IndexableActionableDynamicQuery.class, IndexWriterHelper.class);
 
 	private long _count;
 	private Collection<Document> _documents = new ArrayList<>();
-	private IndexWriterHelper _indexWriterHelper = _indexWriterHelperProxy;
+	private IndexWriterHelper _indexWriterHelper =
```

Comment:
@JanisZhang Is this field really needed? It has a related setter method, which is only used in unit test. We should at least remove the setter method, and double check whether we need to keep this a field, because as a field, the first get happens upon the first instantiation, while without this field, the first get happens upon the first actual method call.